source/ocsp-support/ocsp-support.rst

```diff
@@ -181,32 +181,48 @@ response.
 MongoClient Configuration
 --------------------------
 
-This specification introduces the client-level configuration option
-defined below. Drivers that can, on a per MongoClient basis, disable
-non-stapled OCSP while keeping stapled OCSP enabled MUST implement
-this option.
+This specification introduces the client-level configuration options
+defined below.
 
 tlsDisableOCSPEndpointCheck
```

Comment:
Apologies for asking a bit out-of-scope, but was this option considered for CRL distribution points as well? AFAICT, some TLS libraries give the granularity to disable network access on both revocation checks, but not fine-grained enough to only disable network checks for OCSP.

- Secure Channel has [SCH_CRED_IGNORE_REVOCATION_OFFLINE](https://docs.microsoft.com/en-us/windows/win32/api/schannel/ns-schannel-schannel_cred), which I believe applies to both CRL and OCSP (though I have yet to manually verify with CRL)
- Secure Transport has a policy constant [kSecRevocationNetworkAccessDisabled](https://developer.apple.com/documentation/security/1563600-revocation_policy_constants?language=objc) which I believe has the same behavior.

Reply:
> does SCH_CRED_IGNORE_REVOCATION_OFFLINE prevent Secure Channel from reaching out to the OCSP/CRL endpoints, or does it merely allow for the OCSP/CRL endpoints to be offline

Ah that's true. I believe it still reaches out, but ignores an error if it fails to reach the responder. So even if `SCH_CRED_IGNORE_REVOCATION_OFFLINE` is set, that does not really satisfy this option since it still likely attempts to reach out.

> If the C driver on Windows using Secure Channel can only disable both OCSP and CRLs, then perhaps it would be a candidate for implementing `tlsDisableCertificateRevocationCheck`?

I was thinking just within the scope of `tlsDisableOCSPEndpointCheck`. We want to still do revocation checking, but not reach out to the network if the response wasn't stapled and we don't have a cached response. Is it ok if our implementation of `tlsDisableOCSPEndpointCheck` also disables reaching out to fetch a CRL list?

Looking at dotnet, I'd imagine `CERT_CHAIN_REVOCATION_CHECK_CACHE_ONLY` has the same behavior, i.e. it disables externally fetching for both CRLs and OCSP.

Either way, think this probably out-of-scope of this PR. Feel free to resolve, and we can continue the conversation elsewhere if need be.

Reply:
Will do! :)